Tribler/Core/Libtorrent/LibtorrentMgr.py

```diff
@@ -204,16 +212,19 @@ def create_session(self, hops=0, store_listen_port=True):
             ltsession.listen_on(self.tribler_session.config.get_anon_listen_port(),
                                 self.tribler_session.config.get_anon_listen_port() + 20)
             ltsession.start_dht()
-
             ltsession_settings = ltsession.get_settings()
             ltsession_settings['upload_rate_limit'] = self.tribler_session.config.get_libtorrent_max_upload_rate()
             ltsession_settings['download_rate_limit'] = self.tribler_session.config.get_libtorrent_max_download_rate()
-            ltsession.set_settings(ltsession_settings)
+            if is_1_1_version:
+                ltsession.apply_settings(ltsession_settings)
+            else:
+                ltsession.set_settings(ltsession_settings)
 
         for router in DEFAULT_DHT_ROUTERS:
             ltsession.add_dht_router(*router)
```

Comment:
it's much better to add these as part of the settings when the session is constructed. (I'll make a note there too)

Reply:
Yup, I completely agree! I previously submitted a PR that did just that (aswell as join the DHT nodes). https://github.com/Tribler/tribler/pull/3084/files#diff-63cb62bb807795516a444dcad4ba3097 but it got rejected. Mostly because it broke older versions. This formulation/structure is the minimal that @devos50 came up with that results in the least extra code to support the various versions.